node/narwhal/src/helpers/storage.rs

```diff
@@ -817,22 +817,258 @@ pub mod tests {
+            (any::<CommitteeContext>(), 0..MAX_GC_ROUNDS)
+                .prop_map(|(CommitteeContext(committee, _), gc_rounds)| {
+                    Storage::<CurrentNetwork>::new(committee, gc_rounds)
+                })
+                .boxed()
+        }
+
+        fn arbitrary_with(context: Self::Parameters) -> Self::Strategy {
+            (Just(context), 0..MAX_GC_ROUNDS)
+                .prop_map(|(CommitteeContext(committee, _), gc_rounds)| {
+                    Storage::<CurrentNetwork>::new(committee, gc_rounds)
+                })
+                .boxed()
+        }
+    }
+
+    use crate::helpers::{now, storage::tests::assert_storage};
+
+    use ::bytes::Bytes;
+    use proptest::{prelude::Arbitrary, test_runner::TestRng};
+    use rand::{CryptoRng, Error, Rng, RngCore};
+
+    use snarkvm::{
+        ledger::narwhal::{Batch, Data},
+        prelude::Signature,
+    };
+
+    // The `proptest::TestRng` doesn't implement `rand_core::CryptoRng` trait which is required in snarkVM, so we use a wrapper
+    #[derive(Debug)]
+    pub struct CryptoTestRng(TestRng);
+
+    impl Arbitrary for CryptoTestRng {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<CryptoTestRng>;
+
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            Just(0).prop_perturb(|_, rng| CryptoTestRng(rng)).boxed()
+        }
+    }
+    impl RngCore for CryptoTestRng {
+        fn next_u32(&mut self) -> u32 {
+            self.0.next_u32()
+        }
+
+        fn next_u64(&mut self) -> u64 {
+            self.0.next_u64()
+        }
+
+        fn fill_bytes(&mut self, dest: &mut [u8]) {
+            self.0.fill_bytes(dest);
+        }
+
+        fn try_fill_bytes(&mut self, dest: &mut [u8]) -> std::result::Result<(), Error> {
+            self.0.try_fill_bytes(dest)
+        }
+    }
+
+    impl CryptoRng for CryptoTestRng {}
+
+    #[derive(Debug, Clone)]
+    pub struct AnyTransmission(pub Transmission<CurrentNetwork>);
+
+    impl Arbitrary for AnyTransmission {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<AnyTransmission>;
+
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            any_transmission().prop_map(AnyTransmission).boxed()
+        }
+    }
+
+    #[derive(Debug, Clone)]
+    pub struct AnyTransmissionID(pub TransmissionID<CurrentNetwork>);
+
+    impl Arbitrary for AnyTransmissionID {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<AnyTransmissionID>;
+
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            any_transmission_id().prop_map(AnyTransmissionID).boxed()
+        }
+    }
+
+    fn any_transmission() -> BoxedStrategy<Transmission<CurrentNetwork>> {
+        prop_oneof![
+            (collection::vec(any::<u8>(), 512..=512))
+                .prop_map(|bytes| Transmission::Solution(Data::Buffer(Bytes::from(bytes)))),
+            (collection::vec(any::<u8>(), 2048..=2048))
```

Comment:
why the fixed sizes? if we don't care about the contents at all, the buffer might as well be empty or have 1 byte

Reply:
Good question. I basically converted the [existing  `sample_transmission`](https://github.com/AleoHQ/snarkOS/blob/5261addfaf42ec28ac0e20abbbb6376b8d87ccd9/node/narwhal/src/helpers/storage.rs#L620-L631)- function into a strategy. 

I can check with @howardwu what's the minimum amount of random data needed to make this random data generation a bit faster, and submit the changes in a separate PR.